### PR TITLE
CDRIVER-3806 fix DNS result size comparison

### PR DESCRIPTION
**Pull request by kevinAlbs**
Includes a test with a small initial buffer size. See the ticket for a description of the issue.

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-client.c [hunk 1]
```diff
@@ -411,14 +411,15 @@ static bool
 _mongoc_get_rr_search (const char *service,
                        mongoc_rr_type_t rr_type,
                        mongoc_rr_data_t *rr_data,
+                       size_t initial_buffer_size,
```
Comment (kevinAlbs) [inline src/libmongoc/src/mongoc/mongoc-client.c:414]: Having this argument is for the sake of testing.

File: src/libmongoc/tests/test-mongoc-dns.c [hunk 2]
```diff
@@ -497,6 +497,30 @@ test_srv_polling_mocked (void *unused)
...
+   mongoc_rr_data_t rr_data;
+   bson_error_t error;
+   /* Size needs to be large enough to fit DNS answer header to not error, but
+    * smaller than SRV response to test. The SRV response is 155 bytes. This can
+    * be determined with: dig -t SRV _mongodb._tcp.test1.test.build.10gen.cc */
+   size_t small_buffer_size = 30;
+
+   memset (&rr_data, 0, sizeof (rr_data));
+   ASSERT_OR_PRINT (
+      _mongoc_client_get_rr ("_mongodb._tcp.test1.test.build.10gen.cc",
```
Comment (kevinAlbs) [inline src/libmongoc/tests/test-mongoc-dns.c:513]: Hard coding this is not fantastic, but it's a part of the initial seedlist discovery test suite and spec tests: https://github.com/mongodb/specifications/tree/master/source/initial-dns-seedlist-discovery/tests

If it does change in the future, there will necessarily be a spec change.

**Review by alcaeus — APPROVED** [review]
LGTM. Yay for consistent APIs across distros.

**Comment by kevinAlbs** [comment]
> Yay for consistent APIs across distros.

Haha it boggles my mind why the man pages or system headers do not document the return value.

**Pull request merged**